utils.c

```diff
@@ -155,9 +156,13 @@ xcalloc (size_t size)
 void *
 xrealloc (void *ptr, size_t size)
 {
-  void *res = realloc (ptr, size);
+  void *res;
 
-  if (size != 0 && res == NULL)
+  assert (ptr == NULL || size != 0);
```

Comment:
This implements "crash with an assertion failure if called with (NULL, 0)", but that's not really the same thing as "check the rest of the source code to make sure nobody calls it with (NULL, 0) at the moment". Did you also do the latter?

Reply:
If it would be easier to check that size is never 0 for any caller, then doing that instead would also be fine (and this assertion could be simplified).